src/k2/httpproxy/HTTPProxy.cpp

```diff
@@ -362,6 +362,125 @@ seastar::future<std::tuple<Status, CollectionCreateResponse>> HTTPProxy::_handle
     });
 }
 
+inline seastar::future<nlohmann::json> JsonResponse(Status&& status) {
+    nlohmann::json resp;
+    resp["status"] = status;
+    return seastar::make_ready_future<nlohmann::json>(std::move(resp));
+}
+
+inline seastar::future<nlohmann::json> JsonResponse(Status&& status, nlohmann::json&& response) {
+    nlohmann::json jsonResponse;
+    jsonResponse["status"] = status;
+    jsonResponse["response"] = std::move(response);
+    return seastar::make_ready_future<nlohmann::json>(std::move(jsonResponse));
+}
+
+// Get FieldToKeyString value for a type
+template <class T> void getEscapedString(SchemaField& field, nlohmann::json& jsonval,  String& out) {
+    T val;
+    (void)field;
+    if constexpr  (std::is_same_v<T, std::decimal::decimal64>
```

Comment:
I think you can remove this condition, anything that FieldToKeyString supports we should support here.

Reply:
It fails even before calling `FieldToKeyString`, when it tries to get value into val variable as  `nlohmann::basic_json<>::get_to(std::decimal::decimal64&)` not defined. I think there is an issue for decimal support.